enhancements/validate-catalog-for-bad-bundles.md

```diff
@@ -0,0 +1,177 @@
+Example PriorityClass.yaml
+```yaml
+apiVersion: scheduling.k8s.io/v1
+kind: PriorityClass
+metadata:
+  name: high-priority
+value: 1000000
+globalDefault: false
+description: "This priority class should be used for XYZ service pods only."
+```
+- `globalDefault` should always be set to false
+
+
+### VerticalPodAutoScaler yaml validation
+Example VerticalPodAutoScaler.yaml
+```yaml
+apiVersion: autoscaling.k8s.io/v1
+kind: VerticalPodAutoscaler
+metadata:
+  name: my-rec-vpa
+spec:
+  targetRef:
+    apiVersion: "apps/v1"
+    kind:       Deployment
+    name:       my-rec-deployment
+  updatePolicy:
+    updateMode: "Off"
+```
+- 
+
+## Proposal
+
+The design proposal has two parts:
+- The first discusses potential approaches for validating SAP Gardener objects.
+- The second part discusses design approach for having a tool to list "bad" bundles in a catalog
+
+## Part 1: Validate bundle objects
+
+### 1. Extending Validation library to add validations in `Golang`
+- Extend [BundleValidator](https://github.com/operator-framework/api/blob/master/pkg/validation/internal/bundle.go#L14) to include new validation functions for each of the SAP Gardener objects. (ToDo: Do we add any checks under [BundleFormat](https://github.com/operator-framework/operator-registry/blob/master/pkg/lib/bundle/validate.go#L65) ?)
+- The validation function for each kube manifest object would check for conditions specified under [Static Validation Rules](https://hackmd.io/7TlDq9t_RuGjGm8YsEbyHA?both#Static-Validation-Rules).
+- (ToDo: 
+    -    Do we want to maintain some sort of `struct` for storing various kube-manifest objects as these objects could grow (in the future) or
+    -    Wait to see how these new objects are represented in the [Bundle](https://github.com/operator-framework/api/blob/master/pkg/manifests/bundle.go#L11) as part SAP gardener PR and then take a call)
+ 
+### 2. Using [CUE](https://cuelang.org/docs/integrations/yaml/#validate-yaml-files)
+- CUE is a configuration language that allows you to define a `<pdb-checks>.cue` template for validating Kube Manifests like PDBs.
+- CUE can be an extendable validation library to accept validate template for a resource like PDBs before applying them on to a cluster.
+
+
+#### Pros
+- Not hinging on a third-party operator like OPA Gatekeeper.
+- Validation is extendable to customized .cue files
+
+#### Cons
+- Not a lot to offer in terms of flexibility as compared to [OpenPolicyAgent (OPA)](https://kubernetes.io/blog/2019/08/06/opa-gatekeeper-policy-and-governance-for-kubernetes/)
+- Require users supplying CUE validation template to be familiar with a new type of language
+
+### 3. Using [KUBEVAL](https://kubeval.instrumenta.dev/) - (yet to investigate)
+
+
+## Part 2: Tool to list "bad" bundles in a catalog
+The basic approach will be to:
+- Extend [`opm`](https://github.com/operator-framework/operator-registry/blob/master/docs/design/opm-tooling.md) tooling to add a new command `opm registry extract`
```

Comment:
nit: If this is an opm command that is interacting with index images and not the underlying database implementation, I think this should be `opm index extract`, and the image should probably just be a positional argument.

Reply:
Got it, although I am a little concerned it would be confused with already existing `opm index export`.  Perhaps `opm index unpack`?